### DISCUSSION TITLE
[Feature] Data replication

### DISCUSSION THREAD

**Discussion by lukew3** [immich-app/immich#7820] [Feature Request]
### The feature

I'd like to have my images stored on Immich to be mirrored so that in case one storage medium is destroyed, I have a copy of that data on a separate server. I think this might look something like [mongoDB replication](https://www.mongodb.com/docs/v2.4/core/replication-introduction/).

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
This isn't something for Immich to handle. Rather, you should handle this on the filesystem level with something like ZFS/RAID and by taking regular automated backups that you store offsite.

**Reply by TheOneValen** [reply]
> This isn't something for Immich to handle. Rather, you should handle this on the filesystem level with something like ZFS/RAID and by taking regular automated backups that you store offsite.

I disaggree. I think this is something that should be a first class feature. Take for example https://ente.io/ which I guess (I use and love immich, have not tried ente) is similar to immich but started with the as-a-service thought (in which direction immich seems to be moving). Ente replicates the data to 3 sites.

I think this would be an awesome feature for immich to have in the future. Where you just point it at 1 or more s3 (azblob etc.) stores and it replicates the data for safekeeping (encrypted of course). If you want it to have it as user friendly as possible this should be part of the package.

**Reply by bo0tzz** [reply]
Backups protect not just against storage failure, but also against things like app bugs. If the app is pointed at 3 storage locations, but it has a bug and deletes a file from all of them, the replication doesn't provide any value. So I maintain that this is better managed outside of Immich.

**Reply by TheOneValen** [reply]
The replication is for availability, correct. One can configure different storage append only etc. For a kind of "backup" (in quotes!)

On the way to even think about such a feature a storage layer abstraction is the first step. I see that there a a few issues open suggesting in that direction already. That would certanly help with many other features as well (high availability, multi-host-deployment etc.)

I am very exicted what the future holds for immich, now that it is backed by FUTO
